File: Python/recursiveBubbleSort.py
def bubbleSort(a, n):
	if n==1:
		return
	for i in range(n-1):
		if a[i] > a[i+1]:
			temp = a[i+1]
			a[i+1] = a[i]
			a[i] = temp
	print(a)
	return bubbleSort(a, n-1)

a = [12, 45, 62, 7, 1, 54, 16, 85, 97, 21]
print(a)
bubbleSort(a, len(a))
print(a)

File: CPP/recursiveBubbleSort.cpp
#include<bits/stdc++.h>
using namespace std;

void bubbleSort(int a[], int n)
{
	if(n==1)
		return;
	int j;
	for(j=0; j<n-1; j++)
		{
			if(a[j]>a[j+1])
			{
				int temp = a[j];
				a[j] = a[j+1];
				a[j+1] = temp;
			}
		}
		
		for(j=0; j<10 ;j++)
			cout << a[j] << " ";
		cout << "\n";
	bubbleSort(a, n-1);
}

int main()
{
	cout << "\n";
	int a[] = {5,6,7,8,4,1,10,25,11,13};
	bubbleSort(a, 10);
	for(int j=0;j<10;j++)
		cout << a[j] << " m ";
	cout << "\n";		
	return 0;
}
